src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -184,5 +185,51 @@ protected override void GetUnusedNamespaceImports(SemanticModel model, HashSet<S
                 }
             }
         }
+
+        private static bool IsTupleInDeconstruction(SyntaxNode tuple)
+        {
+            Contract.Assert(tuple.IsKind(SyntaxKind.TupleExpression));
+            var parent = tuple.Parent;
+            do
+            {
+                if (parent.IsKind(SyntaxKind.SimpleAssignmentExpression))
+                {
+                    return true;
+                }
+            }
+            while ((tuple = parent.Parent).IsKind(SyntaxKind.TupleExpression)
+                && (parent = parent.Parent.Parent) != null);
+
+            return false;
+        }
+
+        // This code is copied from the compiler
```

Comment:
Moved methods to `SyntaxFacts`. I didn't create separate APIs for extracting anonymous types and tuples though.
Note: those APIs will also exist in VB (that's for a later PR).